### PR TITLE
feat: publish eks-addon helm chart

### PR DESCRIPTION
**Pull request by nadav3396**

### CODE REVIEW

**Review by mmanciop — CHANGES_REQUESTED** [review]
This violates DRY big times, don't think it's a sensible approach.

File: .github/workflows/build-test-release.yml [hunk 1]
```diff
@@ -458,11 +458,24 @@ jobs:
           make docker-buildx-telemetry-proxy
 
   publish-helm-chart:
+    strategy:
+      matrix:
+        operator-type: [ regular, eks-addon ]
```
Comment (mmanciop) [inline .github/workflows/build-test-release.yml:463]: `helm-registry`?

Comment (mmanciop) [inline .github/workflows/build-test-release.yml:463]: `public-gallery`

File: .github/workflows/build-test-release.yml [hunk 2]
```diff
@@ -458,11 +458,24 @@ jobs:
...
+        include:
+          - operator-type: regular
+            ecr-registry: public.ecr.aws
+            kube-rbac-proxy-image: gcr.io/kubebuilder/kube-rbac-proxy
+            kube-rbac-proxy-tag: v0.13.0
```
Comment (mmanciop) [inline .github/workflows/build-test-release.yml:468]: This looks very wrong

File: .github/workflows/build-test-release.yml [hunk 3]
```diff
@@ -487,24 +500,32 @@ jobs:
...
+          yq -i '(select(documentIndex == 1).spec.template.spec.containers[] | select(.name == "telemetry-proxy")).image = "${{ matrix.ecr-registry }}/lumigo/lumigo-kubernetes-telemetry-proxy:${{ needs.validate-release-increment.outputs.version }}"' config/manager/manager.yaml
+          yq -i '(select(documentIndex == 1).spec.template.spec.containers[] | select(.name == "kube-rbac-proxy")).image = "${{ matrix.kube-rbac-proxy-image }}:${{ matrix.kube-rbac-proxy-tag }}"' config/default/manager_auth_proxy_patch.yaml
+          yq -i '(select(documentIndex == 1).spec.template.spec.containers[] | select(.name == "manager")).env[] | select(.name == "LUMIGO_INJECTOR_IMAGE")).value = "${{ matrix.ecr-registry }}/lumigo/lumigo-autotrace:latest"' config/manager/manager.yaml
+          yq -i '(select(documentIndex == 1).spec.template.spec.containers[] | select(.name == "manager")).env[] | select(.name == "LUMIGO_INJECTOR_IMAGE")).value = "${{ matrix.ecr-registry }}/lumigo/lumigo-autotrace:latest"' config/default/manager_pin_injector_version_patch.yaml
```
Comment (mmanciop) [inline .github/workflows/build-test-release.yml:516]: What's this?

**Review by mmanciop — CHANGES_REQUESTED** [review]

File: .github/workflows/build-test-release.yml (hunk 2 above)
Comment (nadav3396) [reply .github/workflows/build-test-release.yml:467]: will be fixed at - https://lumigo.atlassian.net/browse/RD-11301?atlOrigin=eyJpIjoiNmQ1NTNmZThjMzYwNGQ0NjljMjgzYTA0MGM2ZWY5MGQiLCJwIjoiaiJ9

**Review by moshe-shaham-lumigo — APPROVED** [review]

**Pull request merged**